ext/standard/uuid.c

```diff
@@ -0,0 +1,662 @@
+		return;
+	}
+
+	if (php_uuid_parse_throw(&uuid, Z_STRVAL_P(input), Z_STRLEN_P(input)) == FAILURE) {
+		return;
+	}
+
+	new_uuid(return_value, &uuid);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_parse_args, 0, 1, self, 0)
+	ZEND_ARG_TYPE_INFO(0, input, IS_STRING, 0)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUID, v3)
+{
+	zval *namespace = NULL;
+	zval *name      = NULL;
+	php_uuid *nsid  = NULL;
+	php_uuid uuid;
+
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "zz", &namespace, &name) == FAILURE) {
+		return;
+	}
+
+	nsid = get_uuid(namespace);
+	if (nsid == NULL) {
+		return;
+	}
+
+	php_uuid_create_v3(&uuid, nsid, Z_STRVAL_P(name), Z_STRLEN_P(name));
+	new_uuid(return_value, &uuid);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_v3_args, 0, 2, self, 0)
+	ZEND_ARG_OBJ_INFO(0, namespace, self, 0)
+	ZEND_ARG_TYPE_INFO(0, name, IS_STRING, 0)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUID, v4)
+{
+	php_uuid uuid;
+
+	if (zend_parse_parameters_none_throw() == FAILURE) {
+		return;
+	}
+
+	if (php_uuid_create_v4_throw(&uuid) == FAILURE) {
+		return;
+	}
+
+	new_uuid(return_value, &uuid);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_v4_args, self, 0)
+ZEND_END_ARG_INFO()
+
+PHP_METHOD(UUID, v5)
+{
+	zval *namespace = NULL;
+	zval *name      = NULL;
+	php_uuid *nsid  = NULL;
+	php_uuid uuid;
+
+	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "zz", &namespace, &name) == FAILURE) {
+		return;
+	}
+
+	nsid = get_uuid(namespace);
+	if (nsid == NULL) {
+		return;
+	}
+
+	php_uuid_create_v5(&uuid, nsid, Z_STRVAL_P(name), Z_STRLEN_P(name));
+	new_uuid(return_value, &uuid);
+}
+ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(UUID_v5_args, 0, 2, self, 0)
+ZEND_ARG_OBJ_INFO(0, namespace, self, 0)
+ZEND_ARG_TYPE_INFO(0, name, IS_STRING, 0)
+ZEND_END_ARG_INFO()
+
+#define PHP_UUID_NAMED_CONSTRUCTOR(php_method_name, c_function_name)            \
+	PHP_METHOD(UUID, php_method_name)                                           \
+	{                                                                           \
+		const php_uuid c_function_name = php_uuid_##c_function_name();          \
+		                                                                        \
+		if (zend_parse_parameters_none_throw() == FAILURE) {                    \
+			return;                                                             \
+		}                                                                       \
+		                                                                        \
+		new_uuid(return_value, &c_function_name);                               \
+	}                                                                           \
+	ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO(UUID_##php_method_name##_args, self, 0) \
+	ZEND_END_ARG_INFO();
+
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceDNS, namespace_dns)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceOID, namespace_oid)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceURL, namespace_url)
+PHP_UUID_NAMED_CONSTRUCTOR(NamespaceX500, namespace_x500)
+PHP_UUID_NAMED_CONSTRUCTOR(Nil, nil)
```

Comment:
I don't think this is worth it.

Reply:
What exactly do you mean? The macro?

Reply:
Yeah, referring to the macro here. It saves a couple lines of code, but will break symbol search etc. As there's only four of them, I think it's better to just write them out.